src/libs/actions/App.ts

```diff
@@ -559,6 +560,7 @@ function clearOnyxAndResetApp(shouldNavigateToHomepage?: boolean) {
             });
         });
     });
+    clearSoundAssetsCache();
```

Comment:
Will not it crash mobile app? It looks like `clearSoundAssetsCache` is implemented only on web and such function doesn't exist on mobile? Correct me if I'm wrong 🙏 

Also, why do we need to clear a cache on logout? In case if we want to replace sound we can just change name and we'll simply download new version of the file? Don't you think that `clearSoundAssetsCache` is kind of over engineering?

Reply:
> Will not it crash mobile app? 

I think it won't crash, there is check `if (!('caches' in window)) {` and also try catch, but let me know if we need to extract the function and create platform specific code

```tsx

function clearSoundAssetsCache() {
    // Exit early if the Cache API is not available in the current browser.
    if (!('caches' in window)) {
        return;
    }

    caches
        .delete('sound-assets')
        .then((success) => {
            if (success) {
                return;
            }
            Log.alert('[sound] Failed to clear sound assets cache.');
        })
        .catch((error) => {
            Log.alert('[sound] Error clearing sound assets cache:', {message: (error as Error).message});
        });

```

> Also, why do we need to clear a cache on logout?

The assets might be replaced without changing the name.

> we can just change name and we'll simply download new version of the file?

replacing the name might need code change to the assets references and we might need to rebuild the app each time we change the name

> Don't you think that clearSoundAssetsCache is kind of over engineering?

I don't think so, the code is quiet simple and it's also make sense to make it as part of app reset

Reply:
> I think it won't crash, there is check if (!('caches' in window)) {

I think `clearSoundAssetsCache` will be undefined and you will receive `undefined is not a function`. Did you test native code in this PR?

> The assets might be replaced without changing the name.

At the moment only developer can update asset. So if they update it without changing a name it'll be kind of developer fault...

> replacing the name might need code change to the assets references and we might need to rebuild the app each time we change the name

We can replace only name of `SOUNDS`, i. e.:

```tsx
const SOUNDS = {
    DONE: 'done-v1',
    SUCCESS: 'success',
    ATTENTION: 'attention',
    RECEIVE: 'receive',
} as const;
```

And other code will be untouched 🤷‍♂️ 

> I don't think so, the code is quiet simple and it's also make sense to make it as part of app reset

Oki doki, just wanted to be sure, that alternative solution also has been considered 👍

Reply:
> I think clearSoundAssetsCache will be undefined and you will receive undefined is not a function.

@kirillzyusko you are right, I've updated the native implementation to empty function. thank you!